File: server/src/main/java/com/tomato/bookstore/model/Advertisement.java
package com.tomato.bookstore.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/** 广告实体类 */
@Entity
@Table(name = "advertisements")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Advertisement {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false, length = 50)
  private String title;

  @Column(nullable = false, length = 500)
  private String content;

  @Column(nullable = false, length = 500)
  private String imageUrl;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "product_id", nullable = false)
  private Product product;
}

File: client/src/types/advertisement.ts
/**
 * 广告数据类型
 */
export interface Advertisement {
  /**
   * 广告 ID
   */
  id?: number;

  /**
   * 广告标题
   */
  title: string;

  /**
   * 广告内容
   */
  content: string;

  /**
   * 广告图片 URL
   */
  imageUrl: string;

  /**
   * 关联商品 ID
   */
  productId: number;
}
